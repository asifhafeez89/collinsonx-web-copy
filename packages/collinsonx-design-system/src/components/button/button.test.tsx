import * as React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Button from '.';

const mockFn = jest.fn();

describe('<Button />', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders outline variant', () => {
    const tree = renderer
      .create(
        <Button
          variant="outline"
          color="dark"
          fullWidth={true}
          handleClick={mockFn}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('gets clicked once ', async () => {
    render(
      <>
        <Button
          variant="outline"
          color="dark"
          fullWidth={true}
          handleClick={mockFn}
        />
      </>
    );

    const button = screen.getByRole('button');

    userEvent.click(button);

    await waitFor(() => {
      expect(mockFn).toHaveBeenCalledTimes(1);
    });
  });
});
