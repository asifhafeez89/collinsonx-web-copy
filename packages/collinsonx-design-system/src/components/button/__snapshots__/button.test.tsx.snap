// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders outline variant 1`] = `
.emotion-2 {
  -webkit-tap-highlight-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  cursor: pointer;
  border: 0;
  padding: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: 1rem;
  background-color: transparent;
  text-align: left;
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  height: 2.25rem;
  padding-left: 1.125rem;
  padding-right: 1.125rem;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  display: block;
  width: 100%;
  border-radius: 0.25rem;
  font-weight: 600;
  position: relative;
  line-height: 1;
  font-size: 0.875rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  border: 0.0625rem solid #25262b;
  background-color: transparent;
  color: #25262b;
}

.emotion-2:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

@media (hover: hover) {
  .emotion-2:hover {
    background-color: rgba(193, 194, 197, 0.35);
  }
}

@media (hover: none) {
  .emotion-2:active {
    background-color: rgba(193, 194, 197, 0.35);
  }
}

.emotion-2:active {
  -webkit-transform: translateY(0.0625rem);
  -moz-transform: translateY(0.0625rem);
  -ms-transform: translateY(0.0625rem);
  transform: translateY(0.0625rem);
}

.emotion-2:disabled,
.emotion-2[data-disabled] {
  border-color: transparent;
  background-color: #e9ecef;
  color: #adb5bd;
  cursor: not-allowed;
  background-image: none;
  pointer-events: none;
}

.emotion-2:disabled:active,
.emotion-2[data-disabled]:active {
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-2[data-loading] {
  pointer-events: none;
}

.emotion-2[data-loading]::before {
  content: "";
  position: absolute;
  top: -0.0625rem;
  right: -0.0625rem;
  left: -0.0625rem;
  bottom: -0.0625rem;
  background-color: rgba(255, 255, 255, .5);
  border-radius: 0.25rem;
  cursor: not-allowed;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  overflow: visible;
}

.emotion-5 {
  white-space: nowrap;
  height: 100%;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
  className="emotion-0 emotion-1 emotion-2"
  data-button={true}
  onClick={[MockFunction]}
  type="button"
>
  <div
    className="emotion-3 emotion-4"
  >
    <span
      className="emotion-5 emotion-6"
      style={
        {
          "textTransform": undefined,
        }
      }
    />
  </div>
</button>
`;
